packages/recommendations-core/src/getRecommendations.ts

```diff
@@ -0,0 +1,178 @@
+              globalHitsPerPage > 0
+                ? Math.ceil(globalHitsPerPage / props.objectIDs.length)
+                : globalHitsPerPage;
+            const searchParametersForModel = getSearchParametersForModel({
+              fallbackFilters: props.fallbackFilters,
+              recommendations,
+              threshold: props.threshold,
+            })(props.model);
+
+            return {
+              indexName: props.indexName,
+              params: {
+                hitsPerPage,
+                ...searchParametersForModel,
+                ...props.searchParameters,
+                facetFilters: ((searchParametersForModel.facetFilters ||
+                  []) as string[]).concat(
+                  (props.searchParameters.facetFilters || []) as string[]
+                ),
+                optionalFilters: ((searchParametersForModel.optionalFilters ||
+                  []) as string[]).concat(
+                  (props.searchParameters.optionalFilters || []) as string[]
+                ),
+              },
+            };
+          })
+        )
+        .then((response) => {
+          const hits =
+            // Since recommendations from multiple indices are returned, we
+            // need to sort them descending based on their score.
+            sortBy<ProductRecord<TObject>>(
+              (a, b) => {
+                const scoreA = a.__recommendScore || 0;
+                const scoreB = b.__recommendScore || 0;
+
+                return scoreA < scoreB ? 1 : -1;
+              },
+              // Multiple identical recommended `objectID`s can be returned b
+              // the engine, so we need to remove duplicates.
+              uniqBy<ProductRecord<TObject>>(
+                'objectID',
+                response.results.flatMap((result) =>
+                  result.hits.map((hit, index) => {
+                    const match = recommendationsList
+                      .flat()
+                      .find((x) => x.objectID === hit.objectID);
+
+                    return {
+                      ...hit,
+                      __indexName: props.indexName,
+                      __queryID: result.queryID,
+                      __position: index + 1,
+                      __recommendScore: match?.score ?? null,
+                    };
+                  })
+                )
+              )
+            ).slice(
+              0,
+              // We cap the number of recommendations because the previously
+              // computed `hitsPerPage` was an approximation due to `Math.ceil`.
+              props.maxRecommendations > 0
+                ? props.maxRecommendations
+                : undefined
+            );
+
+          return {
+            recommendations: props.transformItems(hits),
+          };
+        });
+    })
+    .catch(() => {
+      // The `objectID` doesn't exist, we cannot get recommendations.
+      return {
+        recommendations: [],
+      };
```

Comment:
should this not be an error instead of an empty state? It seems hard to debug

(sorry for commenting on unrelated code just because it renamed)

Reply:
Since we mimic the final API with some workarounds with the Search endpoint, we didn't yet bother processing errors. I think we can rather tackle this more efficiently once we implement the new Recommendations API client spec, and have this discussion then. Wdyt?